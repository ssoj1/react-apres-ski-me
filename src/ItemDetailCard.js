import React from "react";
import { Redirect, useParams } from "react-router-dom";
import { Card, CardBody, CardTitle, CardText } from "reactstrap";

/** Shows details of a menu item.
 * 
 * Props: 
 * - items - will be an array of objects of either snacks or drinks like: 
 * [{item}, ...]
 * - cantFind - a string, either "/snacks" or "/drinks" - to reroute to if an 
 * item doesn't exist
 * 
 * State: 
 * - none
 * 
 * Routes -> ItemDetailCard
 */
function ItemDetailCard({ items, cantFind }) {
  console.debug("* ItemDetailCard ", { items, cantFind });
  
  const { id } = useParams();

  let item = items.find(item => item.id === id);
  if (!item) return <Redirect to={cantFind} />;

  return (
    <section>
      <Card>
        <CardBody>
          <CardTitle className="font-weight-bold text-center">
            {item.name}
          </CardTitle>
          <CardText className="font-italic">{item.description}</CardText>
          <p>
            <b>Recipe:</b> {item.recipe}
          </p>
          <p>
            <b>Serve:</b> {item.serve}
          </p>
        </CardBody>
      </Card>
    </section>
  );
}

export default ItemDetailCard;
