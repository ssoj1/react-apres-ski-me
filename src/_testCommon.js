
// sample data to be used in testing

const snacks = [
    {
        "id": "test-snack",
        "name": "Test Snack",
        "description": "Test snack description.",
        "recipe": "Test snack recipe.",
        "serve": "Test snack serve."
      }
];

const drinks = [
    {
        "id": "test-drink",
        "name": "Test Drink",
        "description": "Test drink description.",
        "recipe": "Test drink recipe.",
        "serve": "Test drink serve."
      }
]; 

export { snacks, drinks };