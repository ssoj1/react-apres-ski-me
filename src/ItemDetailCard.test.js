import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import ItemDetailCard from "./ItemDetailCard";
import { snacks } from "./_testCommon";

test("renders test-snack without crashing", function() {
    jest.mock('react-router-dom', () => ({
        useParams: jest.fn().mockReturnValue("test-snack"),
    }));

    render(
        <MemoryRouter>
            <ItemDetailCard items={snacks} cantFind="/snacks" />
        </MemoryRouter>);
  });

test("matches snacks snapshot", function () {
    jest.mock('react-router-dom', () => ({
        useParams: jest.fn().mockReturnValue("test-snack"),
    }));

  const { asFragment } = render(
      <MemoryRouter>
        <ItemDetailCard items={snacks} cantFind="/snacks" />
      </MemoryRouter>,
  );
  expect(asFragment()).toMatchSnapshot();
});
