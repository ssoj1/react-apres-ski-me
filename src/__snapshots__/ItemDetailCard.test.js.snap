// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snacks snapshot 1`] = `<DocumentFragment />`;
